lessons/03_introduction_to_differential_privacy.qmd

```diff
@@ -0,0 +1,550 @@
+
+Post-processing also provides the opportunity to improve utility. Data stewards can use available public or expert knowledge to reduce the amount of noise without accruing additional privacy loss. The public information can come from data released without formal privacy or from individuals who are comfortable sharing their information without noise.
+
+::: {.callout-tip}
+## A key takeaway
+Formal privacy is transparent and allows users to account for the noise introduced into statistics. Post-processing can give up some of this transparency and make it more difficult to account for the noise added to statistics.
+:::
+
+<br>
+
+## Application: Laplace Mechanism
+
+To enhance our intuition, we are going to apply one of the most basic formally privacy algorithms, the Laplace mechanism. This mechanism satisfies DP.
+
+### Laplace Distribution
+
+As the name implies, the Laplace mechanism samples from the Laplace distribution. If you have not encountered the Laplace distribution, it may be helpful to compare it to a more common distribution: the normal distribution.
+
+- Like the normal distribution, the Laplace distribution has parameters for scale/spread (analogous to $\sigma$ in the normal distribution) and location (analogous to mean in the normal distribution).
+
+- We will only ever adjust the spread of the distribution; the location is always assumed to be centered at 0.
+
+```{r echo = FALSE}
+# package containing Laplace distribution
+library(smoothmest)
+
+# set seed for reproducibility
+set.seed(19042022)
+
+# plot comparison
+ggplot() + 
+  geom_function(fun = function(x) dnorm(x, 0, 1),
+                xlim = c(-5,5),
+                aes(color = 'Normal')) +
+  geom_function(fun = function(x) smoothmest::ddoublex(x, 0, 1),
+                 xlim = c(-5,5),
+                aes(color = 'Laplace')) +
+  theme(axis.text.y=element_blank(),
+        axis.ticks.y=element_blank(),
+        axis.title = element_blank(),
+        panel.grid.major = element_blank()) +
+  labs(title = 'Laplace vs. Normal Distributions',
+       color = 'Distribution')
+```
+
+
+<br>
+
+### Adding Noise
+
+To add differentially private noise to a single statistic using the Laplace distribution, we take one random sample from the distribution where:
+
+- the distribution is centered at 0 and 
+
+- the width of the distribution is the ratio of the sensitivity of the target statistics over the privacy loss budget, $\epsilon$.
+
+The resulting noise is then added to the statistic in question.
+
+# Exercise 2
+
+In this exercise, we're going to practice applying the Laplace mechanism by drawing from the Laplace distribution using varying values of sensitivity and epsilon.
+
+Using the `laplace_mechanism()` function defined above (make sure you have run this code), vary the `epsilon` and `sensitivity` parameters. Some sample code is included below. Since we are not trying to reproduce these results, there is no need to set a seed.
```

Comment:
I don't actually see this defined anywhere. If we're just trying to show the intuition without coding (which I think we should aim for given that we haven't run code so far in this series), should we change up the phrasing?